custom/ilsgateway/tanzania/handlers/zipline.py

```diff
@@ -0,0 +1,64 @@
+from corehq.apps.products.models import SQLProduct
+from custom.ilsgateway.tanzania.handlers.keyword import KeywordHandler
+from custom.ilsgateway.tanzania.reminders import INVALID_PRODUCT_CODE
+from custom.zipline.api import ProductQuantity
+
+
+class ParseError(Exception):
+    pass
+
+
+class ZiplineGenericHandler(KeywordHandler):
+
+    help_message = None
+    error_message = None
+
+    def _check_quantities(self, quantities):
+        for quantity in quantities:
+            try:
+                int(quantity)
+            except ValueError:
+                raise ParseError(self.error_message)
+
+    def _check_product_codes(self, product_codes):
+        for product_code in product_codes:
+            try:
+                SQLProduct.objects.get(code=product_code, domain=self.domain)
+            except SQLProduct.DoesNotExist:
+                raise ParseError(INVALID_PRODUCT_CODE % {'product_code': product_code})
+
+    def parse_message(self, text):
+        text_split = text.split()
+        product_codes = text_split[::2]
+        quantities = text_split[1::2]
```

Comment:
In the SOH handler, you can send a message without spaces e.g. soh ab20 cd50. Also, if you send a message like soh ab 20 cd it will be correctly parsed (ab will be 20, cd will be ignored). Do we want that kind of parsing here?